### PR TITLE
display project members

### PR DESCRIPTION
**Pull request by KristinaGru**
## 📝 Description

- Add query to get first and last user names by their uids
- Display avatars of project members in the project details
- Display full user name on avatar hover

## 🎉 Demos

image

### CODE REVIEW

File: backend/src/routes/user/index.ts [hunk 1]
```diff
@@ -3,13 +3,15 @@ import {
...
   getAllUsers,
+  getUserNamesByUids,
 } from '../../controllers/user';
 import express, { Router } from 'express';
 
 const userRoute: Router = express.Router();
 
 userRoute.get('/:uid', getUserById);
 userRoute.get('/', getAllUsers);
+userRoute.post('/names', getUserNamesByUids);
```
Comment (kralle333) [inline backend/src/routes/user/index.ts:14]: you probably dont want post here?

File: backend/src/controllers/user/index.ts [hunk 2]
```diff
@@ -58,3 +58,16 @@ export const getAllUsers = async (req: Request, res: Response) => {
...
   }
 };
+
+export const getUserNamesByUids = async (req: Request, res: Response) => {
+  const { uids } = req.body;
+  try {
+    const users = await db('users').select('first_name', 'last_name').whereIn('uid', uids);
+
+    const userNames = users.map((user) => `${user.first_name} ${user.last_name}`);
+
```
Comment (kralle333) [inline backend/src/controllers/user/index.ts]: why dont you just return user objects?

File: frontend/src/pages/projectView/ProjectView.tsx [hunk 3]
```diff
@@ -38,6 +39,13 @@ const ProjectView = () => {
     Done: 'done',
   };
 
+  const getUserInitials = (name: string) => {
+    return name
+      .split(' ')
+      .map((word) => word.charAt(0).toUpperCase())
+      .join('');
+  };
```
Comment (kralle333) [inline frontend/src/pages/projectView/ProjectView.tsx:47]: and then use the user objects to print the name nicely

**Review by kralle333 — CHANGES_REQUESTED** [review]
Think you can make the code cleaner by using user objects, even if they only have first and last name and then you can use it however you like in the frontend

File: backend/src/routes/user/index.ts (hunk 1 above)
Comment (KristinaGru) [reply backend/src/routes/user/index.ts:14]: I have initially tried get, but can't seem to figure out how to send the array in get request. Axios does not support data object in the body for get requests, but I also tried putting it in params object, as well as directly as a string in the url params, but whatever I do, best case scenario is that I get an empty response data array

**Review by kralle333 — APPROVED** [review]
much cleaner :) I still feel like its wrong with the `post`, but i guess there is nothing to do about that now ;)

**Comment by KristinaGru** [comment]
Thank you @kralle333! I feel the same, so will try to circle back to it later 🙌

**Pull request merged**